smart-contracts/ethereum/contracts/MarketPlace.sol

```diff
@@ -1,43 +1,105 @@
-pragma solidity ^0.4.15;
-
-/**
- * The MarketPlace contract:
- * - stores all the information about hyperbridge apps
- */
-contract Marketplace {
-  modifier onlyOperator() {
-    require(operators[msg.sender]);
+pragma solidity >=0.4.15;
+
+contract MarketPlace {
+
+  modifier onlyDelegates() {
+    require(delegates[msg.sender]);
     _;
   }
 
-  struct appVersion {
-    uint version;
+  struct AppVersion {
+    bytes32 version;
     string files;
-    string category;
-    string status;
+    mapping (address => int8) votes;
+    address[] voters; // we might exclude old voters after elections
     string checksum;
-    uint required_permissions;
+    uint permissions;
   }
+  
 
   struct App {
     uint id;
     address owner;
     string name;
-    appVersion[] versions;
+    string category;
+    mapping  (bytes32 => AppVersion)  versions;
+    bytes32[] versionList;
   }
   
   mapping (uint => App) apps;
-  mapping (address => bool) operators;
+  uint[] app_ids;
```

Comment:
Why are we mixing underscores and camel case?

Reply:
oops i missed this one. I thought i switched everything over to camel case